actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -240,6 +288,56 @@ def handle_unverified_request
       end
     end
 
+    class SessionStore
+      def fetch(request)
+        request.session[:_csrf_token]
+      end
+
+      def store(request, csrf_token)
+        request.session[:_csrf_token] = csrf_token
+      end
+
+      def reset(request)
+        request.session.delete(:_csrf_token)
+      end
+    end
+
+    class CookieStore
+      def initialize(cookie = :csrf_token)
+        @cookie = cookie
+      end
+
+      def fetch(request)
+        contents = request.cookie_jar.encrypted[@cookie]
+        return nil if contents.nil?
+
+        value = JSON.parse(contents).symbolize_keys
```

Comment:
We should probably handle the `JSON` being invalid. Probably handle it like if the cookie was missing.

Reply:
Also I'd drop the `symbolize_keys` since `value` doesn't exit this method, so no point using symbols.